packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -57,7 +58,17 @@ export interface PluginPackage {
     extensionPack?: string[];
 }
 export namespace PluginPackage {
+    function isRunningInElectron(): boolean {
+        const isElectronMain = typeof process !== 'undefined' && process.versions.electron !== undefined;
+        const isElectronRenderer = typeof navigator !== 'undefined' && navigator.userAgent.toLowerCase().includes('electron');
+        const isVsCodeExtensionInElectron = typeof process !== 'undefined' && process.env.ELECTRON_RUN_AS_NODE !== undefined;
+        return (isElectronMain || isElectronRenderer || isVsCodeExtensionInElectron);
+    }
+
     export function toPluginUrl(pck: PluginPackage, relativePath: string): string {
+        if (isRunningInElectron()) {
```

Comment:
We should use `environment` from `'@theia/application-package/lib/environment'`, then `environment.electron.is()`.

Reply:
Thanks for the pointer, though we do need `ELECTRON_RUN_AS_NODE` to detect whether we are in a VS Code extension process. Not sure if it makes sense to add that one to the `environment.electron.is()` check as well?

Reply:
I think it's OK to enhance `is` a little bit; feel free adding `ELECTRON_RUN_AS_NODE`.

Reply:
I'm importing `environment` from `@theia/application-package` since it's re-exported there. Also added `ELECTRON_RUN_AS_NODE` to it though not sure if that could cause unintended side effects.